darts/models/forecasting_model.py

```diff
@@ -644,3 +644,72 @@ def predict(self,
         if self._expect_covariates and covariates is None:
             raise_log(ValueError('The model has been trained with covariates. Some matching covariates '
                                  'have to be provided to `predict()`.'))
+
+
+class ExtendedForecastingModel(ForecastingModel, ABC):
+    """ The base class for "extended" forecasting models, handling optional covariates.
+
+    All implementations have to implement the `fit()` and `predict()` methods defined below.
+    The `fit()` method is meant to train the model on a time series, along with optional
+    covariates.
+    """
+
+    _expect_covariates = False
+
+    @abstractmethod
+    def fit(self,
+            series: TimeSeries,
+            covariates: Optional[Union[TimeSeries]] = None
```

Comment:
I think here it'd just be `Optional[TimeSeries]`

Reply:
You are right! I have changed it.